tests/pr_test.go

```diff
@@ -75,6 +79,85 @@ func setupOptionsQuickStartPattern(t *testing.T, prefix string, dir string) *tes
 	return options
 }
 
+type RecursiveCopy struct {
+	excludeDirs []string
+
+	includeFiletypes []string
+
+	includeDirs []string
+}
+
+func recursiveCopy(dir string, dirsToExclude []string, fileTypesToInclude []string) []string {
+	r := RecursiveCopy{dirsToExclude, fileTypesToInclude, nil}
+	filepath.WalkDir(dir, func(path string, entry fs.DirEntry, err error) error {
+		return walk(&r, path, entry, err)
+	})
+	return r.includeDirs
+}
+
+func walk(r *RecursiveCopy, s string, d fs.DirEntry, err error) error {
+	if err != nil {
+		return err
+	}
+	if d.IsDir() {
+		for _, excludeDir := range r.excludeDirs {
+			if strings.Contains(s, excludeDir) {
+				return err
+			}
+		}
+		if s == ".." {
+			r.includeDirs = append(r.includeDirs, "*.tf")
+			return nil
+		}
+		for _, includeFiletype := range r.includeFiletypes {
+			r.includeDirs = append(r.includeDirs, strings.ReplaceAll(s+"/*"+includeFiletype, "../", ""))
+		}
+	}
+	return nil
+}
+
+func TestRunQuickStartPatternSchematics(t *testing.T) {
+	t.Parallel()
+
+	excludeDirs := []string{
+		".terraform",
+		".docs",
+		".github",
+		".git",
+		".idea",
+		"common-dev-assets",
+		"examples",
+		"tests",
+		"reference-architectures",
+	}
+	includeFiletypes := []string{
+		".tf",
+		".yaml",
+		".py",
+		".tpl",
+	}
+
+	// set up a schematics test
+	options := testschematic.TestSchematicOptionsDefault(&testschematic.TestSchematicOptions{
+		Testing:                t,
+		TarIncludePatterns:     recursiveCopy("..", excludeDirs, includeFiletypes),
```

Comment:
Mentioned above, the recursive routine does not return an error condition to check. Ideally you would call that function above, assign it to a variable, and also check if it errored and fail test.

Something like:
tarIncludes, recurseErr := recursiveCopy(blah blah)
if assert.NoError(t, recurseErr, "some description") {
  return
}